Documentation/architecture.rst

```diff
@@ -147,6 +147,63 @@ single node Cilium deployments used for basic testing / learning, Cilium can
 use a local store implemented as a golang hash map, avoiding the need to setup
 a dedicated KV store.
 
+******
+Labels
+******
+
+Labels are a generic, flexible and highly scaleable way of addressing a large
+set of resources as they allow for arbitrary grouping and creation of sets.
+Whenever something needs to be descried, addressed or selected this is done
+based on labels:
+
+- Endpoints are assigned labels as derived from container runtime or the
+  orchestration system.
+- Network policies select endpoints based on labels and allow consumers based
+  on labels.
+- Network policies themselves are described and addressed by labels.
+
+Basic Label: Key/Value Pair
+---------------------------
+
+A label is a pair of strings consisting of a ``key`` and ``value``. A label can
+be formatted as a single string with the format ``key=value``. The key portion
+is mandatory and must be unique. This is typically achieved by using the
+reverse domain name notion, e.g. ``io.cilium.mykey=myvalue``. The value portion
+is optional and can be omitted, e.g. ``io.cilium.mykey``.
+
+Key names should typically consist of the character set ``[a-z0-9-.]``.
+
+When using labels to select resources, both the key and the value must match,
+e.g. when a policy should be applied to all endpoints will label
+``my.corp.foo`` then the label ``my.corp.foo=value`` will not match the
+selector.
+
+Label Source
+------------
+
+A label can be derived from various sources. For example, a Cilium endpoint can
+derive the labels associated to a container by the local container runtime in
+addition to the labels associated with the pod as provided by Kubernetes. The
+container runtime is not aware of the orchestration system which may result in
+possibly conflicting key names.
+
+To resolve this potential conflict, Cilium attached a label source prefix to
+all label keys in the form of ``[source:]key=value``. The source becomes part
+of the key and can be matched at such, e.g. selecting the label
+``k8s:role=frontend`` will only match on the label ``role=frontend`` if the
+label was derived from Kubernetes, it would not match if the label was derived
+from the container runtime. If the source of a label is to be ignored, the
+label can be prefixed with ``any:`` in which case all labels, regardless of the
+source will match as long as the key and value part match.
```

Comment:
Where will the user would use `any:`? Can he start a container with `--label any:foo=bar`? Can he specify `any:` in the policy? Can he use the `policy trace` tool with the source `any:`?

There are lots of new concepts for a first time reader, I think providing a clear example what he should or should not do helps.